src/plugins/intel_gpu/src/kernel_selector/kernels/sdpa/sdpa_kernel_micro.cpp

```diff
@@ -758,9 +784,9 @@ clKernelData SDPAKernelMicro::get_kernel_data(const sdpa_params& params, bool is
 
         kernel.params.arguments.push_back({ArgumentDescriptor::Types::INTERNAL_BUFFER, 3}); // paged attention helper buffer
     } else {
-        if (params.inputs.size() >= 4)
+        if (params.inputs.size() >= 4 && !params.conf.has_const_attn_mask_val)
             kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 3}); // mask
-        if (params.inputs.size() >= 5)
+        if (params.inputs.size() >= 5 && !params.conf.has_const_scale_val)
             kernel.params.arguments.push_back({ArgumentDescriptor::Types::INPUT, 4}); // Scale
```

Comment:
If the mask input was skipped (because it's scalar), but scale is still runtime value, then we need to check input.size() >= 4 and refer to 3d input arg instead of 4th?

Reply:
Updated, thanks!